packages/teams-js/src/public/authentication.ts

```diff
@@ -159,46 +128,25 @@ export namespace authentication {
   }
 
   function authenticateHelper(apiVersionTag: string, authenticateParameters: AuthenticateParameters): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
-      if (
-        GlobalVars.hostClientType === HostClientType.desktop ||
-        GlobalVars.hostClientType === HostClientType.android ||
-        GlobalVars.hostClientType === HostClientType.ios ||
-        GlobalVars.hostClientType === HostClientType.ipados ||
-        GlobalVars.hostClientType === HostClientType.macos ||
-        GlobalVars.hostClientType === HostClientType.rigel ||
-        GlobalVars.hostClientType === HostClientType.teamsRoomsWindows ||
-        GlobalVars.hostClientType === HostClientType.teamsRoomsAndroid ||
-        GlobalVars.hostClientType === HostClientType.teamsPhones ||
-        GlobalVars.hostClientType === HostClientType.teamsDisplays ||
-        GlobalVars.hostClientType === HostClientType.surfaceHub
-      ) {
-        // Convert any relative URLs into absolute URLs before sending them over to the parent window.
-        const link = document.createElement('a');
-        link.href = authenticateParameters.url;
-        // Ask the parent window to open an authentication window with the parameters provided by the caller.
-        resolve(
-          sendMessageToParentAsync<[boolean, string]>(apiVersionTag, 'authentication.authenticate', [
-            link.href,
-            authenticateParameters.width,
-            authenticateParameters.height,
-            authenticateParameters.isExternal,
-          ]).then(([success, response]: [boolean, string]) => {
-            if (success) {
-              return response;
-            } else {
-              throw new Error(response);
-            }
-          }),
-        );
-      } else {
-        // Open an authentication window with the parameters provided by the caller.
-        authHandlers = {
-          success: resolve,
-          fail: reject,
-        };
-        openAuthenticationWindow(authenticateParameters);
```

Comment:
Please make sure @ydogandjiev and Teams know about this. Yuri mentioned that in the past this code path was needed on some browsers and not others because some browsers decided to block the popup from happening on web if it was launched from Teams instead of from inside TeamsJS. He mentioned that Firefox and Safari specifically were a problem, but would be good to do a thorough test pass of all major browsers to make sure this change isn't going to break anything.

Reply:
Sure, that's the plan before checking in the code.

Reply:
I have created a document and share it with Yuri.